src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -2,277 +2,930 @@
+{
+    public virtual object P { get; set; }
+}
+public class Derived : Base
+{
+    public override string P { get => string.Empty; }
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
+                // (8,28): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override string P { get; }
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "P").WithArguments("covariant returns").WithLocation(8, 28)
+                );
+            VerifyOverride(comp, "Derived.P", "System.Object Base.P { get; set; }");
+            VerifyOverride(comp, "Derived.get_P", "System.Object Base.P.get");
+            comp = CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+            VerifyOverride(comp, "Derived.P", "System.Object Base.P { get; set; }");
+            VerifyOverride(comp, "Derived.get_P", "System.Object Base.P.get");
+        }
+
+        [Fact]
+        public void CovariantReturns_14()
+        {
+            var source = @"
+public class Base
+{
+    public virtual object P { get; set; }
+}
+public class Derived : Base
+{
+    public override string P { get => string.Empty; }
+}
+public class Derived2 : Derived
+{
+    public override string P { get => string.Empty; set { } }
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
+                // (8,28): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override string P { get => string.Empty; }
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "P").WithArguments("covariant returns").WithLocation(8, 28),
+                // (12,53): error CS0546: 'Derived2.P.set': cannot override because 'Derived.P' does not have an overridable set accessor
+                //     public override string P { get => string.Empty; set { } }
+                Diagnostic(ErrorCode.ERR_NoSetToOverride, "set").WithArguments("Derived2.P.set", "Derived.P").WithLocation(12, 53)
+                );
+            VerifyOverride(comp, "Derived.P", "System.Object Base.P { get; set; }");
+            VerifyOverride(comp, "Derived.get_P", "System.Object Base.P.get");
+            VerifyOverride(comp, "Derived2.P", "System.String Derived.P { get; }");
+            VerifyOverride(comp, "Derived2.get_P", "System.String Derived.P.get");
+            VerifyNoOverride(comp, "Derived2.set_P");
+            comp = CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                // (12,53): error CS0546: 'Derived2.P.set': cannot override because 'Derived.P' does not have an overridable set accessor
```

Comment:
>// (12,53): error CS0546: 'Derived2.P.set': cannot override because 'Derived.P' does not have an overridable set accessor [](start = 16, length = 121)

I think this diagnostics is confusing because it implies that either there is no setter, or it is not virtual/overridable. I think this is because we are detecting this problem in the "wrong" place.

Reply:
This is an existing diagnostic for this identical scenario.  I am not attempting to improve it in the context of this set of changes.

---
In reply to: [416949513](https://github.com/dotnet/roslyn/pull/43673#discussion_r416949513) [](ancestors = 416949513)

Reply:
> This is an existing diagnostic for this identical scenario.  I am not attempting to improve it in the context of this set of changes.

I think we should improve that because before one could almost never run into this existing diagnostics and it would most like mean what it says. For this scenario, the diagnostic should be improved to explain what is wrong in a way the people can understand. Please, either ad a PROTOTYPE comment or open an issue for this.

---
In reply to: [416983969](https://github.com/dotnet/roslyn/pull/43673#discussion_r416983969) [](ancestors = 416983969,416949513)

Reply:
Recorded as [https://github.com/dotnet/roslyn/issues/43794](https://github.com/dotnet/roslyn/issues/43794)

---
In reply to: [417027167](https://github.com/dotnet/roslyn/pull/43673#discussion_r417027167) [](ancestors = 417027167,416983969,416949513)